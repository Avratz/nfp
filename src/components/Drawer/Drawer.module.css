.nav {
  width: 100%;
  height: 100vh;
  position: fixed;
  top: 0;
  left: 0;
  background-color: var(--neutral-8);
  color: var(--neutral-1);
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  z-index: 90;
  will-change: transform;
}

.navInner {
  width: 100%;
  padding-left: 2rem;
  padding-right: 2rem;
  display: flex;
  flex-direction: column;
  gap: 1rem;
}

.heading,
.button {
  color: var(--neutral-1);
}

.link {
  font-size: var(--text-xl);
  line-height: var(--line-xl);
}

.link,
.link:visited {
  color: var(--neutral-3);
}

.link:hover {
  color: var(--neutral-1);
}

.lineSeparation {
  width: 100%;
  height: 2px;
  margin: 1rem 0;
  background-color: var(--neutral-3);
}
@media (min-width: 1200px) {
  .nav {
    display: none;
  }
}
