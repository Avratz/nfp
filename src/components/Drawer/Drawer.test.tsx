import { render } from '@testing-library/react'
import { Drawer } from './Drawer'

jest.mock('@/ui/Icon/Icon', () => {
  function Icon() {
    return <div>u</div>
  }

  Icon.displayName = 'Icon'
  return {
    __esModule: true,
    Icon,
  }
})

describe('Drawer', () => {
  it('renders Drawer closed', () => {
    const { container } = render(<Drawer isOpen={false} />)
    expect(container).toMatchSnapshot()
  })

  it('renders Drawer open', () => {
    const { container } = render(<Drawer isOpen />)
    expect(container).toMatchSnapshot()
  })
})
