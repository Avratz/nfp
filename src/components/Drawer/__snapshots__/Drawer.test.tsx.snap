// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer renders Drawer closed 1`] = `
<div>
  <nav
    class="nav"
    style="clip-path: inset(0% 0% 100% 0%);"
  >
    <div
      class="navInner"
    >
      <h3
        class="heading 2xl heading"
        style="opacity: 0; transform: translateY(20px) translateZ(0);"
      >
        Menu
      </h3>
      <a
        class="link link"
        href="/"
        style="opacity: 0; transform: translateY(20px) translateZ(0);"
      >
        Discover
      </a>
      <a
        class="link link"
        href="/"
        style="opacity: 0; transform: translateY(20px) translateZ(0);"
      >
        What we do
      </a>
      <div
        class="lineSeparation"
        style="opacity: 0; transform: translateY(20px) translateZ(0);"
      />
      <h3
        class="heading 2xl heading"
        style="opacity: 0; transform: translateY(20px) translateZ(0);"
      >
        Connect
      </h3>
      <a
        class="link button primaryDark base button"
        href="/"
        style="opacity: 0; transform: translateY(20px) translateZ(0);"
      >
        Connect Wallet
      </a>
    </div>
  </nav>
</div>
`;

exports[`Drawer renders Drawer open 1`] = `
<div>
  <nav
    class="nav"
    data-projection-id="1"
    style="clip-path: inset(0% 0% 0% 0%);"
  >
    <div
      class="navInner"
      data-projection-id="2"
    >
      <h3
        class="heading 2xl heading"
        style="opacity: 1; transform: none;"
      >
        Menu
      </h3>
      <a
        class="link link"
        data-projection-id="4"
        href="/"
        style="opacity: 1; transform: none;"
      >
        Discover
      </a>
      <a
        class="link link"
        data-projection-id="5"
        href="/"
        style="opacity: 1; transform: none;"
      >
        What we do
      </a>
      <div
        class="lineSeparation"
        data-projection-id="6"
        style="opacity: 1; transform: none;"
      />
      <h3
        class="heading 2xl heading"
        style="opacity: 1; transform: none;"
      >
        Connect
      </h3>
      <a
        class="link button primaryDark base button"
        data-projection-id="8"
        href="/"
        style="opacity: 1; transform: none;"
      >
        Connect Wallet
      </a>
    </div>
  </nav>
</div>
`;
